### ISSUE TITLE
BatchNorm2d tracks the biased variance in running_var, CPU and CUDA disagree

### ISSUE THREAD

**Issue by CanReader** [CanReader/FastNN#11]
`batch_norm2d` computes the population variance (divide by n) and `update_running`
blends that same value into the running estimate. PyTorch normalises with the
biased variance but tracks the unbiased one (`* n/(n-1)`), because the running
value is an estimate of the population variance from a sample.

```rust
let bn = BatchNorm2d::new(1);
bn.forward(&Tensor::from_vec(vec![1.0, 2.0, 3.0, 4.0], &[4, 1, 1, 1]));
bn.running_var.value().to_vec();
// [1.025]    0.9*1 + 0.1*1.25    (biased)
// PyTorch:   [1.0667]            0.9*1 + 0.1*1.6667
```

Two consequences: a train/eval gap that gets worse as the batch shrinks, and
checkpoints that don't line up with PyTorch.

The annoying part is that our own CUDA kernel already does it right,
`kernels.cu:1360` applies the `n/(n-1)` correction. So the CPU and GPU paths
disagree with each other.

Files: `src/nn/norm.rs:121`, `src/tensor/ops/norm.rs:95`
